### DISCUSSION TITLE
Refreshing context screen

### DISCUSSION THREAD

**Discussion by j3s5e4** [pwndbg/pwndbg#1164] [General]
I've noticed pwndbg doesn't refresh the screen each time I step an instruction like gef does, as in the register context isn't lined up with the top of the screen, it just keeps whatever fits. Is there a way to clear the screen each time I step through a program?

**Comment by gsingh93** [comment]
Can you provide a screenshot of what pwndbg currently shows for you and how it looks in gef?

**Reply by j3s5e4** [reply]
Sure. Here you can see where pwndbg just appends to the last line on the terminal, whereas gef appears to clear the terminal before displaying the context of the next instruction:

![pwndbg](https://user-images.githubusercontent.com/13514555/192159253-0c091073-007e-4caf-8c3d-90d42bc66527.png)
.
![gef](https://user-images.githubusercontent.com/13514555/192159256-30f0c1b4-8963-4b7f-b9d8-a4942c1e3506.png)

**Reply by gsingh93** [reply]
Seems like this should be easy to implement: https://github.com/hugsy/gef/blob/dev/gef.py#L3649. Can you create an issue for this?

**Reply by j3s5e4** [reply]
https://github.com/pwndbg/pwndbg/issues/1165

**Reply by disconnect3d** [reply]
As noted below, this is already implemented in Pwndbg but has to be set via `set context-clear-screen on`. However, we found out in #1165 that there was a small bug with it which was hopefully fixed in https://github.com/pwndbg/pwndbg/commit/3b2f7796d663d1c9562265a86aa81105069129f7

**Comment by stnevans** [comment]
For anyone in the future, `set context-clear-screen on` does this.
